### PR TITLE
Unit vector direction reconstruction

### PR DESCRIPTION
**Pull request by nickwp**
Note that this PR depends on #31 so that should be merged first and then there will be much fewer/simpler changes here.

This adds the ability to predict direction unit vectors instead of zenith and azimuth angles, by adding extra `"directions"` key to the dataset. Augmentation transform functions are updated to also correctly transform the unit vector direction values.
Corresponding updates to the analysis code to properly read and analyse the reasults are included.
The prediction is not forced to be a unit vector during model training or evaluation (although the loss function will encourage it to be) but during analysis the prediction is normalised to a unit vector.

### CODE REVIEW

**Comment by pdeperio** [comment]
change the base to `regression` then?

**Comment by nickwp** [comment]
Since #31 has been merged, this is now ready for review. It has a few extra additions related to improving direction reconstruction: - A fix for output normalisation scaling, so that the target values are scaled during training, then when saving the predictions the inverse scaling is applied to the model output.
- Added feature to choose GroupNorm instead of BatchNorm, since BatchNorm sometimes behaves erratically. But in practice found it still works best.
- Also included a small fix for plotting training progression plots when there's only one epoch.

**Review by pdeperio — APPROVED** [review]
